crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -155,6 +155,13 @@ pub struct Sphere {
 }
 impl Primitive3d for Sphere {}
 
+impl Default for Sphere {
```

Comment:
This is just a side note and doesn't really need addressing in this PR... but it's really weird to me that `Default` is required for `Meshable` as it forces all these defaults that I think really go against the spirit of `Default`.

Reply:
wanna ask this just because I wanna try trackling https://github.com/bevyengine/bevy/issues/11746 and want to understand idiomatic Bevy/Rust -- 
Are you just saying that we shouldn't be using the `Default` trait within each `Meshable::mesh()` implementation to fill in the rest of each respective builder? Not really understanding what how the current impl goes against how `Default` should be used (I'm just new)

Reply:
My personal opinion is that defaults for shapes don't really make sense because there's no particularly opinionated reason to have, say a radius 0.5 sphere instead of radius 1.0 as a default.

But Jondolf makes a good point that they're useful for prototyping and editors. So I've been convinced that the defaults should stay.

Reply:
yeah I was thinking like just based on the `3d_shapes` example from the file changes that it looks p useful as-is ?

Reply:
> ...but it's really weird to me that Default is required for Meshable...

I just double-checked, and `Default` is *not* required for `Meshable` from what I can see.

Reply:
@valenotary

> Are you just saying that we shouldn't be using the Default trait within each Meshable::mesh() implementation to fill in the rest of each respective builder? Not really understanding what how the current impl goes against how Default should be used (I'm just new)

Based on spectria's comment, I thought the `Meshable` trait required all implementors to also have `Default` implemented, i.e. it's defined like `pub trait Meshable: Default { ... }`. In my comment, I was just saying that it probably shouldn't be a strict requirement, because e.g. a `Polygon` might not have a default value that makes much sense.

But it looks to me like this is already the case, and `Default` is not required for `Meshable` (but it should still be implemented for most primitives imo).

And yeah, I think it looks good the way it is.